sdk/python/feast/infra/materialization/snowflake_engine.py

```diff
@@ -271,32 +271,68 @@ def _materialize_one(
 
             fv_latest_values_sql = offline_job.to_sql()
 
+            # Determine payload size if writing to an external online store
+            if self.repo_config.online_store.type != "snowflake.online":
+                if feature_view.entity_columns:
+                    join_keys = [entity.name for entity in feature_view.entity_columns]
+                    unique_entities = '"' + '", "'.join(join_keys) + '"'
+
+                    query = f"""
+                    SELECT
+                        COUNT(*)
+                    FROM
+                    (
+                        SELECT
+                            {unique_entities}
+                        FROM
+                            {feature_view.batch_source.get_table_query_string()}
+                        QUALIFY ROW_NUMBER() OVER (PARTITION BY {unique_entities} ORDER BY {unique_entities}) = 1
```

Comment:
why do you need a window function here? Cant you do like APPROX_COUNT_DISTINCT(unique_entities) or something?

Reply:
updated

Reply:
ideally we do DISTINCT not APPROx